InternTrack.Core.Api/Services/Foundations/Interns/InternService.Exceptions.cs

```diff
@@ -34,6 +34,10 @@ private async ValueTask<Intern> TryCatch(ReturningInternFuction returningInternF
             {
                 throw CreateAndLogValidationException(invalidInternException);
             }
+            catch (NotFoundInternException NotfoundInternException)
```

Comment:
`NotFoundException` shouldn't come before `SqlException` - honor the chronological order here